### PR TITLE
Issue #9510: updated example of AST for TokenTypes.LPAREN

### PR DESCRIPTION
**Pull request by stonefix**
Closes: #9510 

image

```
$ cat Test.java
public class Test {

    void foo(){
        Integer val = new Integer();
        while (false) {
            val += (-3);
        }
    }

}

$ java -jar checkstyle-8.40-all.jar -t Test.java  

CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--METHOD_DEF -> METHOD_DEF [3:4]
    |   |--MODIFIERS -> MODIFIERS [3:4]
    |   |--TYPE -> TYPE [3:4]
    |   |   `--LITERAL_VOID -> void [3:4]
    |   |--IDENT -> foo [3:9]
    |   |--LPAREN -> ( [3:12]
    |   |--PARAMETERS -> PARAMETERS [3:13]
    |   |--RPAREN -> ) [3:13]
    |   `--SLIST -> { [3:15]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [4:8]
    |       |   |--MODIFIERS -> MODIFIERS [4:8]
    |       |   |--TYPE -> TYPE [4:8]
    |       |   |   `--IDENT -> Integer [4:8]
    |       |   |--IDENT -> val [4:16]
    |       |   `--ASSIGN -> = [4:20]
    |       |       `--EXPR -> EXPR [4:22]
    |       |           `--LITERAL_NEW -> new [4:22]
    |       |               |--IDENT -> Integer [4:26]
    |       |               |--LPAREN -> ( [4:33]
    |       |               |--ELIST -> ELIST [4:34]
    |       |               `--RPAREN -> ) [4:34]
    |       |--SEMI -> ; [4:35]
    |       |--LITERAL_WHILE -> while [5:8]
    |       |   |--LPAREN -> ( [5:14]
    |       |   |--EXPR -> EXPR [5:15]
    |       |   |   `--LITERAL_FALSE -> false [5:15]
    |       |   |--RPAREN -> ) [5:20]
    |       |   `--SLIST -> { [5:22]
    |       |       |--EXPR -> EXPR [6:16]
    |       |       |   `--PLUS_ASSIGN -> += [6:16]
    |       |       |       |--IDENT -> val [6:12]
    |       |       |       |--LPAREN -> ( [6:19]
    |       |       |       |--UNARY_MINUS -> - [6:20]
    |       |       |       |   `--NUM_INT -> 3 [6:21]
    |       |       |       `--RPAREN -> ) [6:22]
    |       |       |--SEMI -> ; [6:23]
    |       |       `--RCURLY -> } [7:8]
    |       `--RCURLY -> } [8:4]
    `--RCURLY -> } [10:0]



```
expected update for javodoc is:

 ```
CLASS_DEF -> CLASS_DEF
|--MODIFIERS -> MODIFIERS
|   `--LITERAL_PUBLIC -> public
|--LITERAL_CLASS -> class
|--IDENT -> Test
`--OBJBLOCK -> OBJBLOCK
    |--LCURLY -> {
    |--METHOD_DEF -> METHOD_DEF
    |   |--MODIFIERS -> MODIFIERS
    |   |--TYPE -> TYPE
    |   |   `--LITERAL_VOID -> void
    |   |--IDENT -> foo
    |   |--LPAREN -> (
    |   |--PARAMETERS -> PARAMETERS
    |   |--RPAREN -> )
    |   `--SLIST -> {
    |       |--VARIABLE_DEF -> VARIABLE_DEF
    |       |   |--MODIFIERS -> MODIFIERS
    |       |   |--TYPE -> TYPE
    |       |   |   `--IDENT -> Integer
    |       |   |--IDENT -> val
    |       |   `--ASSIGN -> =
    |       |       `--EXPR -> EXPR
    |       |           `--LITERAL_NEW -> new
    |       |               |--IDENT -> Integer
    |       |               |--LPAREN -> (
    |       |               |--ELIST -> ELIST
    |       |               `--RPAREN -> )
    |       |--SEMI -> ;
    |       |--LITERAL_WHILE -> while
    |       |   |--LPAREN -> (
    |       |   |--EXPR -> EXPR
    |       |   |   `--LITERAL_FALSE -> false
    |       |   |--RPAREN -> )
    |       |   `--SLIST -> {
    |       |       |--EXPR -> EXPR
    |       |       |   `--PLUS_ASSIGN -> +=
    |       |       |       |--IDENT -> val
    |       |       |       |--LPAREN -> (
    |       |       |       |--UNARY_MINUS -> -
    |       |       |       |   `--NUM_INT -> 3
    |       |       |       `--RPAREN -> )
    |       |       |--SEMI -> ;
    |       |       `--RCURLY -> }
    |       `--RCURLY -> }
    `--RCURLY -> }
```

### CODE REVIEW

**Comment by romani** [comment]
@stonefix , please make sure "mvn verify" is passing on your local.
CI should be green.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1732,6 +1732,73 @@
     /**
      * A left parenthesis ({@code (}).
      *
+     * <p>For example:</p>
+     * <pre>
+     *   String str = new String();
+     *   for (int i = 0; i &lt; 2; i++) {
+     * 	        str += (i*((5 + 7) - 3));
+     *   }
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1740]: lets make it more compact to focus on `(` as much as possible:
```
        Integer val = new Integer();
        while (false) {
            val += (-3);
        }
```

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (stonefix) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: @romani Thanks for review and advice. I updated this, but two tests failed.

**Comment by romani** [comment]
Content is good, Please squash all commits in one to pass CI and merge PR

**Pull request closed without merging**